internal/policy/policy.go

```diff
@@ -198,6 +198,26 @@ func LoadStateForEntry(ctx context.Context, repo *git.Repository, e rsl.EntryTyp
 	return state, nil
 }
 
+// GetStateForCommit scans the RSL to identify the first time a commit was seen
+// in the repository. The policy preceding that RSL entry is returned as the
+// State to be used for verifying the commit's signature.
+func GetStateForCommit(ctx context.Context, repo *git.Repository, commit *object.Commit) (*State, error) {
+	firstSeenEntry, err := rsl.GetFirstEntryForCommit(repo, commit)
+	if err != nil {
+		if errors.Is(err, rsl.ErrNoRecordOfCommit) {
+			return nil, nil // should err be set here to indicate it's never been seen?
```

Comment:
Either seems reasonable, but make sure to document this in the godoc.

Reply:
Will do!